jenkins/jobs/capm3-e2e-tests.pipeline

```diff
@@ -0,0 +1,105 @@
+import java.text.SimpleDateFormat
+import java.security.SecureRandom
+
+ci_git_credential_id = "metal3-jenkins-github-token"
+
+// 3 hours
+int TIMEOUT = 10800
+
+KEEP_TEST_ENV = (env.KEEP_TEST_ENV)
+
+script {
+    echo "Checkout ${ghprbAuthorRepoGitUrl} main"
+    ci_git_branch = 'main'
+    ci_git_url = 'https://github.com/metal3-io/project-infra.git'
+
+    int rand = new SecureRandom()
+    VM_KEY = (1..4).collect {
+        ('a'..
+      'z').join('')[rand.nextInt(26)]
+  }.join('')
+
+    agent_label = "metal3ci-large-${IMAGE_OS}"
+}
+
+pipeline {
+    agent none
+    environment {
+        METAL3_CI_USER = 'metal3ci'
+        REPO_ORG = "${PROJECT_REPO_ORG}"
+        REPO_NAME = "${PROJECT_REPO_NAME}"
+        UPDATED_REPO = "${ghprbAuthorRepoGitUrl}"
+        REPO_BRANCH = "${ghprbTargetBranch}"
+        UPDATED_BRANCH = "${ghprbActualCommit}"
+        BUILD_TAG = "${env.BUILD_TAG}"
+        IMAGE_OS = "${IMAGE_OS}"
+        CAPM3RELEASEBRANCH = "${capm3_release_branch}"
+        BMORELEASEBRANCH = "${bmo_release_branch}"
+        TARGET_NODE_MEMORY = "${TARGET_NODE_MEMORY}"
+        NUM_NODES = 4
```

Comment:
I believe it NUM_NODES should be 5 for clusterctl_upgrade tests.

Reply:
yes, but we change it again on CAPM3 side, I will just remove it from here

Reply:
sure. Since we are planning to use it for all e2e tests it makes sense.